plugins/modules/nd_multi_cluster_connectivity.py

```diff
@@ -0,0 +1,289 @@
+description:
+- This module allows for the management of clusters on Cisco Nexus Dashboard.
+- This module is only supported on ND v4.1 and later.
+author:
+- Shreyas Srish (@shrsr)
+options:
+  cluster_type:
+    description:
+    - The type of the cluster.
+    type: str
+    choices: [ nd, apic ]
+  cluster_hostname:
+    description:
+    - The hostname or IP address of the cluster.
+    type: str
+  cluster_username:
+    description:
+    - The username for authenticating with the cluster.
+    type: str
+  cluster_password:
+    description:
+    - The password for authenticating with the cluster.
+    - This value is not logged in the output.
+    type: str
+  cluster_login_domain:
+    description:
+    - The login domain for the cluster.
+    type: str
+  multi_cluster_login_domain:
+    description:
+    - The multi-cluster login domain.
+    type: str
+  fabric_name:
+    description:
+    - The name of the fabric to which the cluster belongs.
+    type: str
+  license_tier:
+    description:
+    - The license tier for the cluster.
+    type: str
+    choices: [ advantage, essentials, premier ]
+  features:
+    description:
+    - A list of features to enable on the cluster.
+    type: list
+    elements: str
+    choices: [ telemetry, orchestration ]
+  inband_epg:
+    description:
+    - The in-band EPG (Endpoint Group) for the cluster.
+    type: str
+  outband:
+    description:
+    - The out-of-band management configuration for the cluster.
+    type: str
+  orchestration:
+    description:
+    - Enable or disable orchestration for the cluster.
+    type: bool
+  security_domain:
+    description:
+    - The security domain for the cluster.
+    type: str
+  validate_peer_certificate:
+    description:
+    - Whether to validate the peer's SSL/TLS certificate.
+    type: bool
+  state:
+    description:
+    - Use C(present) to create or update a cluster configuration.
+    - Use C(absent) to delete a cluster configuration.
+    - Use C(query) to retrieve cluster configurations.
+    type: str
+    choices: [ absent, present, query ]
+    default: present
+extends_documentation_fragment:
+- cisco.nd.modules
+- cisco.nd.check_mode
+"""
+
+EXAMPLES = r"""
+- name: Create ND cluster
+  cisco.nd.nd_multi_cluster_connectivity:
+    cluster_type: nd
+    cluster_hostname: cluster-IP
+    cluster_username: admin
+    cluster_password: cluster-password
+    state: present
+
+- name: Connect an ACI cluster with features
+  cisco.nd.nd_multi_cluster_connectivity:
+    cluster_type: apic
+    fabric_name: test_aci
+    cluster_hostname: cluster-IP
+    cluster_username: admin
+    cluster_password: cluster-password
+    license_tier: premier
+    features:
+      - orchestration
+    inband_epg: ansible-inband
+    state: present
+
+- name: Query ND cluster
+  cisco.nd.nd_multi_cluster_connectivity:
+    cluster_type: nd
+    cluster_hostname: cluster-IP
+    state: query
+  register: query_result
+
+- name: Query all the clusters
```

Comment:
Implement separate query all for ND clusters and aci clusters in the info module